### PR TITLE
Define log levels in docs and change default to 0-3.

### PR DESCRIPTION
**Pull request by elric1**
We define the meaning of the various log levels in the man page
for krb5_openlog(3).  If logging configured and levels are not
specified, we change the default levels to 0-3 which should exclude
debugging messages which are generally only desired in exceptional
circumstances.

We also go through the KDC and adjust the levels to be appropriate.

### CODE REVIEW

File: kdc/connect.c [hunk 1]
```diff
@@ -831,13 +831,13 @@ handle_tcp(krb5_context context,
         }
     } else if (d[idx].len > 4) {
 	kdc_log (context, config,
-		 0, "TCP data of strange type from %s to %s/%d",
+		 1, "TCP data of strange type from %s to %s/%d",
```
Comment (vdukhovni) [inline kdc/connect.c]: Perhaps also a 2 here. Garbage coming in is not our error?

File: kdc/connect.c [hunk 2]
```diff
@@ -831,13 +831,13 @@ handle_tcp(krb5_context context,
...
 		 d[idx].addr_string, descr_type(d + idx),
 		 ntohs(d[idx].port));
 	if (d[idx].buf[0] & 0x80) {
 	    krb5_data reply;
 
-	    kdc_log (context, config, 0, "TCP extension not supported");
+	    kdc_log (context, config, 1, "TCP extension not supported");
```
Comment (vdukhovni) [inline kdc/connect.c]: Same here?

File: kdc/kerberos5.c [hunk 3]
```diff
@@ -358,7 +358,7 @@ pa_pkinit_validate(kdc_request_t r, const PA_DATA *pa)
     ret = _kdc_pk_rd_padata(r->context, r->config, &r->req, pa, r->client, &pkp);
     if (ret || pkp == NULL) {
 	ret = KRB5KRB_AP_ERR_BAD_INTEGRITY;
-	_kdc_r_log(r, 5, "Failed to decode PKINIT PA-DATA -- %s",
+	_kdc_r_log(r, 1, "Failed to decode PKINIT PA-DATA -- %s",
```
Comment (vdukhovni) [inline kdc/kerberos5.c]: Perhaps a 2 here also.

File: kdc/kerberos5.c [hunk 4]
```diff
@@ -476,7 +476,7 @@ pa_enc_chal_validate(kdc_request_t r, const PA_DATA *pa)
 			       &size);
     if (ret) {
 	ret = KRB5KRB_AP_ERR_BAD_INTEGRITY;
-	_kdc_r_log(r, 5, "Failed to decode PA-DATA -- %s",
+	_kdc_r_log(r, 1, "Failed to decode PA-DATA -- %s",
```
Comment (vdukhovni) [inline kdc/kerberos5.c]: Ditto.

File: kdc/kerberos5.c [hunk 5]
```diff
@@ -525,7 +525,7 @@ pa_enc_chal_validate(kdc_request_t r, const PA_DATA *pa)
 	    ret2 = krb5_enctype_to_string(r->context, k->key.keytype, &str);
 	    if (ret2)
 		str = NULL;
-	    _kdc_r_log(r, 5, "Failed to decrypt ENC-CHAL -- %s "
+	    _kdc_r_log(r, 1, "Failed to decrypt ENC-CHAL -- %s "
```
Comment (vdukhovni) [inline kdc/kerberos5.c]: Same.

File: kdc/kerberos5.c [hunk 6]
```diff
@@ -542,7 +542,7 @@ pa_enc_chal_validate(kdc_request_t r, const PA_DATA *pa)
 	if(ret){
 	    krb5_crypto_destroy(r->context, challangecrypto);
 	    ret = KRB5KDC_ERR_PREAUTH_FAILED;
-	    _kdc_r_log(r, 5, "Failed to decode PA-ENC-TS_ENC -- %s",
+	    _kdc_r_log(r, 1, "Failed to decode PA-ENC-TS_ENC -- %s",
```
Comment (vdukhovni) [inline kdc/kerberos5.c]: Same.

File: kdc/kerberos5.c [hunk 7]
```diff
@@ -556,7 +556,7 @@ pa_enc_chal_validate(kdc_request_t r, const PA_DATA *pa)
 			     client_time, sizeof(client_time), TRUE);
 
 	    ret = KRB5KRB_AP_ERR_SKEW;
-	    _kdc_r_log(r, 0, "Too large time skew, "
+	    _kdc_r_log(r, 1, "Too large time skew, "
```
Comment (vdukhovni) [inline kdc/kerberos5.c]: Ditto.

File: kdc/kerberos5.c [hunk 8]
```diff
@@ -619,7 +619,7 @@ pa_enc_ts_validate(kdc_request_t r, const PA_DATA *pa)
 			       &len);
     if (ret) {
 	ret = KRB5KRB_AP_ERR_BAD_INTEGRITY;
-	_kdc_r_log(r, 5, "Failed to decode PA-DATA -- %s",
+	_kdc_r_log(r, 1, "Failed to decode PA-DATA -- %s",
```
Comment (vdukhovni) [inline kdc/kerberos5.c]: -"-

File: kdc/kerberos5.c [hunk 9]
```diff
@@ -674,7 +674,7 @@ pa_enc_ts_validate(kdc_request_t r, const PA_DATA *pa)
 				      pa_key->key.keytype, &str);
 	if (ret2)
 	    str = NULL;
-	_kdc_r_log(r, 5, "Failed to decrypt PA-DATA -- %s "
+	_kdc_r_log(r, 1, "Failed to decrypt PA-DATA -- %s "
```
Comment (vdukhovni) [inline kdc/kerberos5.c]: -"-

File: kdc/kerberos5.c [hunk 10]
```diff
@@ -701,7 +701,7 @@ pa_enc_ts_validate(kdc_request_t r, const PA_DATA *pa)
     krb5_data_free(&ts_data);
     if(ret){
 	ret = KRB5KDC_ERR_PREAUTH_FAILED;
-	_kdc_r_log(r, 5, "Failed to decode PA-ENC-TS_ENC -- %s",
+	_kdc_r_log(r, 1, "Failed to decode PA-ENC-TS_ENC -- %s",
```
Comment (vdukhovni) [inline kdc/kerberos5.c]: Same.

File: kdc/kerberos5.c [hunk 11]
```diff
@@ -712,7 +712,7 @@ pa_enc_ts_validate(kdc_request_t r, const PA_DATA *pa)
 			 client_time, sizeof(client_time), TRUE);
 
 	ret = KRB5KRB_AP_ERR_SKEW;
-	_kdc_r_log(r, 0, "Too large time skew, "
+	_kdc_r_log(r, 1, "Too large time skew, "
```
Comment (vdukhovni) [inline kdc/kerberos5.c]: Same.

**Comment by vdukhovni** [comment]
@elric1 what do you want to do with this one?

**Comment by elric1** [comment]
@vdukhovni: I've adopted your suggestions.

**Pull request merged**